tracer/src/Datadog.Trace/Sampling/TracerRateLimiter.cs

```diff
@@ -18,7 +20,21 @@ public TracerRateLimiter(int? maxTracesPerInterval)
 
         public override void OnDisallowed(Span span, int count, int intervalMs, int maxTracesPerInterval)
         {
-            Log.Warning<string, int, int>("Dropping trace id {TraceId} with count of {Count} for last {Interval}ms.", span.Context.RawTraceId, count, intervalMs);
+            if (!Volatile.Read(ref _warningWritten))
+            {
+                Log.Warning<string, int, int>("Dropping trace id {TraceId} with count of {Count} for last {Interval}ms.", span.Context.RawTraceId, count, intervalMs);
```

Comment:
While you're in here, can we reword this message to make it clearer? It makes it sound like we had `{Count}` trace with `{TraceId}`, which is wrong.

Maybe something like this?

```suggestion
                Log.Warning<string, int, int>("Rate limiter dropped a trace ({TraceId}) after reaching {Count} traces in the last {Interval}ms.", span.Context.RawTraceId, count, intervalMs);
```